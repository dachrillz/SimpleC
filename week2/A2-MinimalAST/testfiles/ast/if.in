int a(){
    if (2 < 4){
        a = 5;
    }
    else{
        b = 2;
    }
    return;
}
