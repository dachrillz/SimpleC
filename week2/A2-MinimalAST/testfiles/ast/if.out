Program
  List
    Function ID="a"
      null
      List
        IfElse
          LT
            Var ID="2"
            Var ID="4"
          List
            IdUse ID="a"
              Var ID="5"
          List
            IdUse ID="b"
              Var ID="2"
        Return
          null
