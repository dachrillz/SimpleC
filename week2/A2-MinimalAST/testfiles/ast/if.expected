Program
  List
    Function ID="a"
      null
      List
        If
          LT
            Numeral NUMERAL="2"
            Numeral NUMERAL="4"
          List
            IdUse ID="a"
              Numeral NUMERAL="5"
          List
            IdUse ID="b"
              Numeral NUMERAL="2"
        Return
